.github/autobuild/ios.sh

```diff
@@ -30,17 +30,76 @@ setup() {
     fi
 }
 
+
+prepare_signing() {
+    [[ "${SIGN_IF_POSSIBLE:-0}" == "1" ]] || return 1
+
+    # Signing was requested, now check all prerequisites:
+    [[ -n "${IOSDIST_CERTIFICATE:-}" ]] || return 1
+    [[ -n "${IOSDIST_CERTIFICATE_ID:-}" ]] || return 1
+    [[ -n "${IOSDIST_CERTIFICATE_PWD:-}" ]] || return 1
+    [[ -n "${NOTARIZATION_PASSWORD:-}" ]] || return 1
+    [[ -n "${IOS_PROV_PROFILE_B64:-}" ]] || return 1
+    [[ -n "${KEYCHAIN_PASSWORD:-}" ]] || return 1
+
+    echo "Signing was requested and all dependencies are satisfied"
+
+    # use this as filename for Provisioning Profile
+    IOS_PP_PATH="embedded.mobileprovision"
+
+    ## Put the cert to a file
+    # IOSDIST_CERTIFICATE - iOS Distribution
+    echo "${IOSDIST_CERTIFICATE}" | base64 --decode > iosdist_certificate.p12
+
+    ## Echo Provisioning Profile to file
+    echo -n "${IOS_PROV_PROFILE_B64}" | base64 --decode > $IOS_PP_PATH
+
+    # Set up a keychain for the build:
+    security create-keychain -p "${KEYCHAIN_PASSWORD}" build.keychain
+    security default-keychain -s build.keychain
+    security unlock-keychain -p "${KEYCHAIN_PASSWORD}" build.keychain
+    security import iosdist_certificate.p12 -k build.keychain -P "${IOSDIST_CERTIFICATE_PWD}" -A -T /usr/bin/codesign
+    # security set-key-partition-list -S apple-tool:,apple:,codesign: -s -k "${KEYCHAIN_PASSWORD}" build.keychain
+    security set-key-partition-list -S apple-tool:,apple: -s -k "${KEYCHAIN_PASSWORD}" build.keychain
+    # set lock timeout on keychain to 6 hours
+    security set-keychain-settings -lut 21600
+    
+    # apply provisioning profile
+    #FIXME - maybe redundant?
+    mkdir -p ~/Library/MobileDevice/Provisioning\ Profiles
+    cp $IOS_PP_PATH ~/Library/MobileDevice/Provisioning\ Profiles
+
+    # for debug
+    echo "Checking found identities..."
+    security find-identity -v 
+
+    # Tell Github Workflow that we need notarization & stapling:
+    echo "::set-output name=macos_signed::true"
```

Comment:
should be renamed to apple_signed if we merge this.

Reply:
or maybe ios_signed

Reply:
@ann0see it _is_ actually possible to upload the binary to the App Store, using "altool" as outlined here: https://help.apple.com/app-store-connect/#/devb1c185036

However, as the App Store submission process involves a number of manual setup steps _anyway_ (ie creating the actual submission on the App Store Connect website, adding screenshots, descriptions, compliance details and all sorts) and you likely only want to upload specific versions to the App Store (rather than having it run automatically), I thought the generation of the signed, verifiable IPA is a big enough win for this automation.

_Quick note: the only manual step remaining once you have the signed IPA is to drag and drop the IPA file into Transporter (on macOS), and click the Upload button._

If I have time today though I will experiment with adding the upload directly using xcrun altool.